pytorch_lightning/trainer/distrib_parts.py

```diff
@@ -517,9 +519,14 @@ def dp_train(self, model):
 
         model.cuda(self.root_gpu)
 
+        # hack forward to do autocast for the user
+        if self.use_amp and self.use_native_amp:
+            model.forward = torch.cuda.amp.autocast()(model.forward)
```

Comment:
@ethanwharris put your suggestion here. see a better place this should be done?

Reply:
Not that I can see, LGTM :)

Reply:
As in the [example](https://pytorch.org/docs/master/notes/amp_examples.html#dataparallel-in-a-single-process), if you intend to wrap with DataParallel, modifying model.forward to enable autocast should occur _before_ wrapping with DataParallel.  Based on code below, it looks like that's what's happening, so this is probably ok here, unless the user already wrapped the model in DataParallel before passing it in.

Reply:
we wrap it for the user. this keeps all the distributed stuff abstracted from their research code.